### PR TITLE
Only the host can change the workflow of a fishbowl

### PR DESCRIPTION
**Pull request by aerrasti**
## Subject
Only the host can change the workflow of a fishbowl

### CODE REVIEW

File: backend/tests/Functional/FishbowlWorkflowFunctionalTest.php [hunk 1]
```diff
@@ -0,0 +1,265 @@
...
+
+        $logInResponse = $response->toArray();
+
+        $this->assertArrayHasKey('token', $logInResponse);
+        $this->assertNotEmpty($logInResponse['token']);
+
+        return $logInResponse['token'];
+    }
+
+    /*** @return array<string, array<string, null|array<string, string>>> */
```
Comment (aerrasti) [inline backend/tests/Functional/FishbowlWorkflowFunctionalTest.php:234]: @jordisala1991 Any help with the phpstan error
`return  
         type has no value type specified in iterable type array.               `

Comment (jordisala1991) [inline backend/tests/Functional/FishbowlWorkflowFunctionalTest.php:234]:
```suggestion
    /**
     * @return array<string, array<string, null|array<string, string>>> 
     */
```

Comment (aerrasti) [reply backend/tests/Functional/FishbowlWorkflowFunctionalTest.php:234]: ` ------ ---------------------------------------------------------------------- 
  Line   tests/Functional/FishbowlWorkflowFunctionalTest.php                   
 ------ ---------------------------------------------------------------------- 
  98     Offset 'currentStatus' does not exist on array<string, string>|null.  
  144    Offset 'currentStatus' does not exist on array<string, string>|null.  
  190    Offset 'currentStatus' does not exist on array<string, string>|null.  
 ------ ---------------------------------------------------------------------- 
`

File: backend/tests/Functional/FishbowlWorkflowFunctionalTest.php [hunk 2]
```diff
@@ -0,0 +1,265 @@
...
+                'variables' => [
+                    'slug' => 'fishbowl-slug',
+                ],
+            ],
+            'auth_bearer' => $token,
+        ]);
+        $graphqlResponse = $response->toArray();
+
+        $this->assertArrayHasKey('data', $graphqlResponse);
+        $this->assertNotEmpty($graphqlResponse['data']);
```
Comment (jordisala1991) [inline backend/tests/Functional/FishbowlWorkflowFunctionalTest.php:249]: Those asserts won't always be true if this pretends to be a generic method for executing queries on graphQL.

File: backend/tests/Functional/FishbowlWorkflowFunctionalTest.php [hunk 3]
```diff
@@ -0,0 +1,265 @@
...
+        $response = static::createClient()->request('POST', '/graphql', [
+            'json' => [
+                'query' => $gql,
+                'variables' => [
+                    'slug' => 'fishbowl-slug',
+                ],
+            ],
+            'auth_bearer' => $token,
+        ]);
+        $graphqlResponse = $response->toArray();
```
Comment (jordisala1991) [inline backend/tests/Functional/FishbowlWorkflowFunctionalTest.php:246]: Wouldn't be better to return just the response? That way you won't have problems with the return array.

Comment (aerrasti) [reply backend/tests/Functional/FishbowlWorkflowFunctionalTest.php:246]: good idea. I can try :)

**Review by jordisala1991 — APPROVED** [review]

**Pull request merged**